crates/meroctl/src/cli/context/create.rs

```diff
@@ -0,0 +1,177 @@
+use calimero_primitives::application::ApplicationId;
+use calimero_primitives::identity::Context;
+use calimero_server_primitives::admin::ApplicationListResult;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use libp2p::Multiaddr;
+use reqwest::Client;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use tracing::info;
+
+use crate::cli::context::common::multiaddr_to_url;
+use crate::cli::RootArgs;
+use crate::config_file::ConfigFile;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateContextRequest {
+    application_id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateContextResponse {
+    data: Context,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InstallApplicationRequest {
+    pub application: ApplicationId,
+    pub version: Version,
+    pub path: Utf8PathBuf,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ListApplicationsResponse {
+    data: ApplicationListResult,
+}
+
+#[derive(Debug, Parser)]
+#[group(requires_all(&["dev", "path", "version"]))]
+pub struct CreateCommand {
+    /// The application ID to attach to the context
+    #[clap(long, short = 'a', exclusive = true, value_name = "APP_ID")]
+    application_id: String,
+
+    /// Enable dev mode
+    #[clap(long, group = "dev_args")]
+    dev: bool,
+
+    /// Path to use in dev mode
+    #[clap(short, long, group = "dev_args", value_name = "PATH")]
+    path: Utf8PathBuf,
+
+    /// Version of the application
+    #[clap(
+        short,
+        long,
+        group = "dev_args",
+        default_value = "0.0.0",
+        value_name = "VERSION"
+    )]
+    version: Version,
```

Comment:
`version` is only useful when `--dev` is provided, the definition should reflect that relationship